framework/packages/abstract-std/src/objects/gov_type.rs

```diff
@@ -117,20 +122,64 @@ impl GovernanceDetails<String> {
                 })
             }
             GovernanceDetails::Renounced {} => Ok(GovernanceDetails::Renounced {}),
+            GovernanceDetails::NFT {
+                collection_addr,
+                token_id,
+            } => {
+                // get owner of token_id from collection
+                let owner: OwnerOfResponse = deps.querier.query_wasm_smart(
+                    &collection_addr,
+                    &cw721::Cw721QueryMsg::OwnerOf {
+                        token_id: token_id.clone(),
+                        include_expired: None,
+                    },
+                )?;
+                // verify owner
+                // if *sender == owner.owner {
+                //     Ok(())
+                // } else {
+                //     Err(ownership_error)
+                // }
+
+                Ok(GovernanceDetails::NFT {
+                    collection_addr: deps.api.addr_validate(&collection_addr.to_string())?,
+                    token_id,
+                })
+            }
         }
     }
 }
 
 impl GovernanceDetails<Addr> {
     /// Get the owner address from the governance details
-    pub fn owner_address(&self) -> Option<Addr> {
+    pub fn owner_address(&self, deps: Option<Deps>) -> Option<Addr> {
```

Comment:
let this take `&Api` instead of `Option<deps>`. Passing a reference that ends up unused is cheap.